frontend/src/hooks/useTerminal.ts

```diff
@@ -131,10 +134,17 @@ export const useTerminal = (commands: Command[] = []) => {
     if (terminal.current && commands.length > 0) {
       // Start writing commands from the last command index
       for (let i = lastCommandIndex.current; i < commands.length; i += 1) {
-        const command = commands[i];
-        terminal.current?.writeln(parseTerminalOutput(command.content));
+        // eslint-disable-next-line prefer-const
+        let { content, type } = commands[i];
 
-        if (command.type === "output") {
+        secrets.forEach((secret) => {
```

Comment:
**Suggestion:** Ensure that the `secrets` array is not empty before processing.

**Existing Code:**
```TypeScript
secrets.forEach((secret) => {
    const secretRegex = new RegExp(secret, "g");
    content = content.replace(secretRegex, "*".repeat(10));
});
```

**Improved Code:**
```TypeScript
if (secrets.length > 0) {
    secrets.forEach((secret) => {
        const secretRegex = new RegExp(secret, "g");
        content = content.replace(secretRegex, "*".repeat(10));
    });
}
```

**Details:**
Add a check to skip the replacement logic if `secrets` is empty to avoid unnecessary processing.